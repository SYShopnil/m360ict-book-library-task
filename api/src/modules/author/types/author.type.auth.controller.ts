export interface IAuthorLoginBOdy {
  password: string;
  email: string;
}
export interface IPayloadDataOfAccessToken {
  id: number;
  email: string;
}
