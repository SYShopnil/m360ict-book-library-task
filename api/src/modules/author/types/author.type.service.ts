import { IAuthor } from '../../../type/entity';

export interface IGetAllAuthors {
  authors: IAuthor[];
  total: number;
  totalPages: number;
}

export interface IRGetAuthorById extends IAuthor {
  email: string;
}
export interface IRGetAuthorByEmail extends IAuthor {
  email: string;
  password: string;
}
