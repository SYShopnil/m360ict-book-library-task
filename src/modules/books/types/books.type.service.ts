import { IBook } from '../../../type/entity';

export interface GetBooksOptions {
  page: number;
  pageSize: number;
  title?: string;
  author?: string;
}

export interface IRGetBookById extends IBook {}
export interface IRGetBooksByAuthor extends IBook {
  author_name: string;
  authors_bio: string;
}
