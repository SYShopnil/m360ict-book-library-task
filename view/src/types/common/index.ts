export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  rUserList = "rUserList",
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CProductSearchBarContainer = "CProductSearchBarContainer",
}
