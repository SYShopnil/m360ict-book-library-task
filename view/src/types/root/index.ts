export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
export type { IButton, INormalButton, IRedirectButton } from "./Button";
export { BtnColorSchema } from "./Button";
