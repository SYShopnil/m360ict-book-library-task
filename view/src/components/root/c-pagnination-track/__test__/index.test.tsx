import { render, screen } from "@testing-library/react";
import { CPaginationTrack } from "..";
import { EDataTestId } from "@src/types/common";

describe("When Pagination Tracker rendered ", () => {
  it("Expect it should render successfully", async () => {
    const { getByTestId } = render(
      <CPaginationTrack currentPage={2} totalPage={5} />
    );
    // const myElement = getByTestId(EDataTestId.CPaginationTrack);
    const myElement = screen.getByTestId(EDataTestId.CPaginationTrack);
    expect(myElement).toMatchSnapshot();
  });
});
