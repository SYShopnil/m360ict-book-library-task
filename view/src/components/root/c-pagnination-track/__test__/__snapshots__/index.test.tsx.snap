// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Pagination Tracker rendered  Expect it should render successfully 1`] = `
<div
  class="flex space-x-2"
  data-testid="CPaginationTrack"
>
  <div
    class="false duration-[0.2s] h-[2rem] w-[2rem]  bg-[#7F4D4F] hover:bg-[#E0E3EA] flex justify-center items-center"
  >
    <span
      class="cursor-pointer false duration-[0.2s]  text-white hover:text-[#7F4D4F]  font-bold text-lg"
    >
      1
    </span>
  </div>
  <div
    class="!bg-[#E0E3EA] duration-[0.2s] h-[2rem] w-[2rem]  bg-[#7F4D4F] hover:bg-[#E0E3EA] flex justify-center items-center"
  >
    <span
      class="cursor-pointer  !text-[#7F4D4F] duration-[0.2s]  text-white hover:text-[#7F4D4F]  font-bold text-lg"
    >
      2
    </span>
  </div>
  <div
    class="false duration-[0.2s] h-[2rem] w-[2rem]  bg-[#7F4D4F] hover:bg-[#E0E3EA] flex justify-center items-center"
  >
    <span
      class="cursor-pointer false duration-[0.2s]  text-white hover:text-[#7F4D4F]  font-bold text-lg"
    >
      3
    </span>
  </div>
  <div
    class="false duration-[0.2s] h-[2rem] w-[2rem]  bg-[#7F4D4F] hover:bg-[#E0E3EA] flex justify-center items-center"
  >
    <span
      class="cursor-pointer false duration-[0.2s]  text-white hover:text-[#7F4D4F]  font-bold text-lg"
    >
      4
    </span>
  </div>
  <div
    class="false duration-[0.2s] h-[2rem] w-[2rem]  bg-[#7F4D4F] hover:bg-[#E0E3EA] flex justify-center items-center"
  >
    <span
      class="cursor-pointer false duration-[0.2s]  text-white hover:text-[#7F4D4F]  font-bold text-lg"
    >
      5
    </span>
  </div>
</div>
`;
